clients/go/pkg/zbc/oauthCredentialsProvider.go

```diff
@@ -60,8 +65,10 @@ type OAuthProviderConfig struct {
 	// the environment variable 'ZEEBE_AUTHORIZATION_SERVER_URL'.
 	AuthorizationServerURL string
 	// Cache to read/write credentials from; if none given, defaults to an oauthYamlCredentialsCache instance with the
-	// path '$HOME/.camunda/credentials' as default (can be overriden by 'ZEEBE_CLIENT_CONFIG_PATH')
+	// path '$HOME/.camunda/credentials' as default (can be overridden by 'ZEEBE_CLIENT_CONFIG_PATH')
 	Cache OAuthCredentialsCache
+	// Timeout is the maximum duration of an OAuth request. The default value is 10 seconds
+	Timeout time.Duration
```

Comment:
We also don't make the client timeout configurable although that's not really a configuration since it can be changed between commands. I'm not sure how important it is for the user but I can use an env var for this if you want

Reply:
What speaks against doing that? Is this because with the context issue we will remove this field?

Reply:
It depends if we want the user to be able to set different timeouts for command API and OAuth. They obviously have different time frames so I'd say that it would still make sense to have it. Even if you disable timeouts for the zeebe API, probably you'll still want a small-ish timeout for the oauth endpoint since there's no valid reason for it to take very long. I can add it as a configuration and make an issue to keep the Java client consistent

Reply:
Let's tackle that along with the whole context issue, as we'll probably also want to propagate the context here.